src/lib/keyboardHandlerFilter.ts

```diff
@@ -0,0 +1,23 @@
+import { KeyCombinations, KeyboardHandler } from './accessibility/interfaces'
+import * as keyboardKey from 'keyboard-key'
+
+const keyboardHandlerFilter = (handler: KeyboardHandler, keysCombinations: KeyCombinations[]) => (
+  event: React.KeyboardEvent,
+) => {
+  const filteredKeys = keysCombinations.filter(keysCombinations => {
+    const keyCode = keysCombinations.keyCode
+    const { shiftKey, altKey, metaKey, ctrlKey } = event
+    if (
+      (keysCombinations.altKey && !altKey) ||
+      (keysCombinations.shiftKey && !shiftKey) ||
+      (keysCombinations.metaKey && !metaKey) ||
+      (keysCombinations.ctrlKey && !ctrlKey)
+    ) {
+      return null
```

Comment:
The predicate should return boolean, so `false` here.
You can also use lodash `some` to improve performance, and (nit) put `keyCode` comparison first in the condition for short-circuit evaluation:
```jsx
const { shiftKey, altKey, metaKey, ctrlKey } = event
const isHandled = _.some(keysCombinations, keysCombination => (
    keysCombination.keyCode === keyboardKey.getCode(event)
    (!keysCombination.altKey || altKey) &&
    (!keysCombination.shiftKey || shiftKey) &&
    (!keysCombination.metaKey || metaKey) &&
    (!keysCombination.ctrlKey || ctrlKey) 
  ))
if (isHandled) {
  handler(event)
}
```